docs/BUILD.md

```diff
@@ -0,0 +1,263 @@
+
+s2n-tls can be configured with the following CMake options. Each option can be set by passing a `-D<option>=<value>` flag to CMake.
+- [**`CMAKE_BUILD_TYPE`**](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html): Sets the build type. Some of the possible build types are as follows:
+  - `Release`: Produce an optimized s2n-tls library artifact with no debug info. This option should be used when building s2n-tls for use in production.
+  - `Debug`: Produce an unoptimized library artifact with debug info. This option can be used when developing for or with s2n-tls. The debug symbols produced with this build can be used with GDB and other utilities to help with debugging.
+- [**`CMAKE_INSTALL_PREFIX`**](https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html): Specifies where the s2n-tls library artifacts are placed when installing s2n-tls.
+- [**`CMAKE_PREFIX_PATH`**](https://cmake.org/cmake/help/latest/variable/CMAKE_PREFIX_PATH.html): Specifies install locations used by CMake to search for library dependencies. This option can be used to link s2n-tls to a specific libcrypto. See the [Building with a specific libcrypto](#building-with-a-specific-libcrypto) section for more information on building with different libcryptos.
+- [**`BUILD_SHARED_LIBS`**](https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html): Specifies whether a static or shared s2n-tls library artifact will be produced during the build. Defaults to `OFF`, building a static library. If set to `ON`, a shared library will be produced instead.
+
+The entire list of s2n-tls CMake options can be viewed with the following command:
+
+```bash
+cmake . -LH
+```
+
+See the [s2n-tls usage guide](USAGE-GUIDE.md#consuming-s2n-tls-via-cmake) for instructions on how to include s2n-tls in your CMake project.
+
+## Building with a specific libcrypto
+
+s2n-tls has a dependency on a libcrypto library. A supported libcrypto must be linked to s2n-tls when building. The following libcrypto libraries are currently supported:
+- [AWS-LC](https://github.com/aws/aws-lc)
+- [OpenSSL](https://www.openssl.org/) (versions 1.0.2 - 3.0)
+  - ChaChaPoly is not supported before Openssl-1.1.1.
+  - RSA-PSS is not supported before Openssl-1.1.1.
+  - RC4 is not supported with Openssl-3.0 or later.
+- [BoringSSL](https://boringssl.googlesource.com/boringssl)
+  - OCSP features are not supported with BoringSSL.
+- [LibreSSL](https://www.libressl.org/)
+
+By default, s2n-tls will attempt to find a system libcrypto to link with when building. However, this search can be overridden to any of the above libcryptos by specifying the install directory with the `CMAKE_PREFIX_PATH` flag.
+
+For help building a desired libcrypto on your platform, please consult the project's build documentation.
+
+### AWS-LC
+
+[AWS-LC](https://github.com/aws/aws-lc) is the recommended libcrypto to use with s2n-tls due to increased performance and security. AWS-LC can be built with the following:
```

Comment:
Should we write out instructions for how to build with aws-lc, or link to existing instructions maintained by aws-lc? I'm not sure we should own this